devito/interfaces.py

```diff
@@ -18,8 +56,23 @@ def set_initializer(self, lambda_initializer):
         assert(callable(lambda_initializer))
         self.initializer = lambda_initializer
 
+    # function to generate a hased file name used for memmap file
+    def _generate_memmap_filenmae(self):
```

Comment:
Why do we not just let a user input a filename/filepath, with a sensible default like `"/tmp/devito-memmap/data_%s" % self.name`?